MapleServer2/Types/Levels.cs

```diff
@@ -114,5 +118,20 @@ public void GainPrestigeExp(long amount)
             Player.Session.Send(PrestigePacket.ExpUp(Player, amount));
         }
 
+        public void GainMasteryExp(MasteryType type, long amount)
+        {
+            Mastery mastery = Masteries.FirstOrDefault(x => x.Type == (byte) type);
+
+            if (mastery == null) // add mastery to list
+            {
+                Masteries.Add(new Mastery(type, amount));
+                Player.Session.Send(MasteryPacket.SetExp(type, amount));
+            }
+            else
+            {
+                // user already has some exp in mastery, so simply update it
+                Player.Session.Send(MasteryPacket.SetExp(type, mastery.CurrentExp += amount));
+            }
+        }
```

Comment:
Should be in the `Levels` class

Reply:
Already is in `Levels` class?!

Reply:
No, I mean this method should be in the Levels class.

Reply:
Whoops lol